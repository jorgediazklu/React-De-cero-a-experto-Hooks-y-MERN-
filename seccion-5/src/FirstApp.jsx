import React from "react";
import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <div>
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </div>
  );
};
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};
FirstApp.defaultProps = {
  name: "No hay nombre",
  //   title: "No hay titulo",
  subTitle: "No hay subtitulo",
};
