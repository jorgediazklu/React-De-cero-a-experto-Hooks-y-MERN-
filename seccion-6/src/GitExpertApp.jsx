import React, { useState } from "react";
import { AddCategories, GifGrid } from "./components";

export const GitExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);
  const onAddCategory = (inputValue) => {
    if (categories.includes(inputValue)) return;
    setCategories((prev) => [...prev, inputValue]);
  };
  return (
    <div>
      GitExpertApp
      <br />
      <br />
      <br />
      <AddCategories onAddCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
