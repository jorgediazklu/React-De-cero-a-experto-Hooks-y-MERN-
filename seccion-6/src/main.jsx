import React from "react";
import ReactDOM from "react-dom/client";
import { GitExpertApp } from "./GitExpertApp";
import "./styles.css";
// pGyq9JNgrL4MkrvIY2TUBlpYOglFFs17
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GitExpertApp />
  </React.StrictMode>
);
