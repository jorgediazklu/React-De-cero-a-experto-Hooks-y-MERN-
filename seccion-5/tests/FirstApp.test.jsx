import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("prueba a <FirstApp/>", () => {
  test("match snapshot", () => {
    const title = "titulo";
    const { container } = render(<FirstApp title={title} />);

    //lo que hace es crear el codigo html directamente en una carpeta llamada __snapshot__
    expect(container).toMatchSnapshot();
  });
  test("match h1", () => {
    const title = "titulo";
    const { container } = render(<FirstApp title={title} />);
    const h1 = container.querySelector("h1");
    //toContain: esto es para saber si esta el titulo y tabien que no de error si hay un espacio entre el inicio o el final de la etiqueta
    expect(h1).toContain(title);
  });
  test("match text", () => {
    const title = "titulo";
    const { container, getByText } = render(<FirstApp title={title} />);

    //getByText: que busque el text que se le pasa como parametro
    //toBeTruthy: que exsista o que si se renderize
    expect(getByText(title)).toBeTruthy();
  });
});
