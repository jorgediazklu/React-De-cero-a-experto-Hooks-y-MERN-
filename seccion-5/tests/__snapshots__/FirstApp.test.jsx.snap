// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba a <FirstApp/> match snapshot 1`] = `
<div>
  <div>
    <h1>
      titulo
    </h1>
    <p>
      No hay subtitulo
    </p>
    <p>
      No hay nombre
    </p>
  </div>
</div>
`;
